.github/workflows/ci-build.yaml

```diff
@@ -176,4 +176,4 @@ jobs:
         pip install -r docs/requirements.txt
     - name: Render documentation
       run: |
-        sphinx-build --color -W --keep-going -b html -D nb_execution_mode=off docs docs/build/html
+        sphinx-build --color -W --keep-going -b html docs docs/build/html
```

Comment:
Turning on notebook execution on GH actions will make that CI run a bit slower, but it shouldn't have as much of a queuing issue as the Read the Docs builds.

Reply:
Can you quantify how much slower? Or rather what's the end to end time?

I say this because we have a very strict 10 minute time limit for CI builds. You cannot exceed that.

Also if there are more notebooks that are added in the future, that might regress this restriction.

Reply:
Good question! It looks like it increased the end to end time from ~5 minutes to ~9 minutes, which does get close to that 10 minute limit, so we could turn that off again, but I do think it would be good to execute the notebooks _somewhere_! I guess it's a question about where is the best place. Maybe we could set it up as part of an internal workflow? I'm only sketching this as an option for handling some of out Read the Docs issues, but I don't really have strong opinions about this part :D

Reply:
9 minutes is close. And with more notebooks being added, this will indeed go past the limit. Any CI build that blocks PRs/CLs have to respect the 10 minute time limit.

Read the docs is an option. Let's wait to see if we can get on the better plan there and this problem would be solved.

Reply:
Sure - I think it's missing the point a bit, to focus on this change which isn't the key point of this idea (which I think still has some merit to consider even if we do get a RTDs Pro plan!), but like I said, I'm not too fussed either way...

Reply:
Under 10 is under 10. I wouldn't overly worry about a hypothetical future overage.

A couple of possibilities are:
a) reintroduce parallelism in the sphinx build and use a larger runner, and
b) only build the notebooks if a notebook changed in the PR.

Reply:
Well, the point of me commenting on this PR was to let Dan know about the time limit we enforce. I have no opinions on how you want to solve this problem as long as the blocking CI build stays under 10 minutes. I agree that we shouldn't worry about future changes.

Reply:
Thanks both! I'll come back to this one a bit later to look into the options here if we decide that something like this is useful, but for now, I'll add a bit more info to the description above.